pkg/api/pod/util.go

```diff
@@ -571,11 +571,6 @@ func dropDisabledFields(
 		})
 	}
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.PodOverhead) && !overheadInUse(oldPodSpec) {
-		// Set Overhead to nil only if the feature is disabled and it is not used
-		podSpec.Overhead = nil
```

Comment:
all the PodOverhead stuff looks like a bad rebase?

Reply:
Yes. Sorry, I'm switching back and forth between too many different branches at the moment.

PR updated.